components/tidb_query_executors/src/limit_executor.rs

```diff
@@ -225,7 +231,7 @@ mod tests {
             ],
         );
 
-        let mut exec = BatchLimitExecutor::new(src_exec, 4).unwrap();
+        let mut exec = BatchLimitExecutor::new(src_exec, 4, false).unwrap();
```

Comment:
Please also consider adding some cases for `is_src_scan_executor = true`.

Reply:
add test_src_exec_is_scan